addon/models/school.js

```diff
@@ -1,75 +1,61 @@
-  templatePrefix: attr('string'),
-  iliosAdministratorEmail: attr('string'),
-  changeAlertRecipients: attr('string'),
-  competencies: hasMany('competencies', { async: true }),
-  courses: hasMany('course', { async: true }),
-  programs: hasMany('program', { async: true }),
-  vocabularies: hasMany('vocabulary', { async: true }),
-  instructorGroups: hasMany('instructor-group', { async: true }),
-  curriculumInventoryInstitution: belongsTo('curriculum-inventory-institution', { async: true }),
-  sessionTypes: hasMany('session-type', { async: true }),
-  directors: hasMany('user', { async: true, inverse: 'directedSchools' }),
-  administrators: hasMany('user', {
-    async: true,
-    inverse: 'administeredSchools',
-  }),
-  configurations: hasMany('school-config', { async: true }),
+export default class School extends Model {
+  @attr('string')
+  title;
+  @attr('string')
+  templatePrefix;
+  @attr('string')
+  iliosAdministratorEmail;
+  @attr('string')
+  changeAlertRecipients;
+  @hasMany('competencies', { async: true })
+  competencies;
+  @hasMany('course', { async: true })
+  courses;
+  @hasMany('program', { async: true })
+  programs;
+  @hasMany('vocabulary', { async: true })
+  vocabularies;
+  @hasMany('instructor-group', { async: true })
+  instructorGroups;
+  @belongsTo('curriculum-inventory-institution', { async: true })
+  curriculumInventoryInstitution;
+  @hasMany('session-type', { async: true })
+  sessionTypes;
+  @hasMany('user', { async: true, inverse: 'directedSchools' })
+  directors;
+  @hasMany('user', { async: true, inverse: 'administeredSchools' })
+  administrators;
+  @hasMany('school-config', { async: true })
+  configurations;
 
-  cohorts: computed('id', 'programs.@each.programYears', 'store', {
-    get() {
-      return this.store.query('cohort', {
-        filters: {
-          schools: [this.id],
-        },
-      });
-    },
-  }).readOnly(),
-
-  async getCohortsForYear(year) {
-    const cohorts = await this.cohorts;
-    const cohortsForYear = await filter(cohorts.toArray(), async (cohort) => {
-      const programYear = await cohort.get('programYear');
-      const classOfYear = await programYear.get('classOfYear');
-      const start = parseInt(programYear.get('startYear'), 10);
-      const end = parseInt(classOfYear, 10);
-      return parseInt(year, 10) >= start && parseInt(year, 10) <= end;
-    });
-    return cohortsForYear;
-  },
-
-  /**
-   * Get all program years starting in the given year.
-   * @method getProgramYearsForYear
-   * @param {Number} year
-   * @return {Promise.<Array>}
-   */
-  async getProgramYearsForYear(year) {
-    const programs = await this.programs;
-    const rhett = await map(programs.mapBy('programYears'), (programYears) => {
-      return programYears.filter((programYear) => {
-        return parseInt(programYear.get('startYear'), 10) === year;
-      });
+  get cohorts() {
+    deprecate(`school.cohorts is called, use getCohorts() instead.`, false, {
```

Comment:
I think we should encourage ourselves to just get the cohorts by loading everything, pulling the cohorts off the store guarantees a network request even if the data is already loaded. My preference would be for this getter to return the `store.query` promise from `getCohorts` and for us to stop using it ASAP.

Reply:
updated :heavy_check_mark: